src/vs/gitpod/node/gitpodInsightsAppender.ts

```diff
@@ -0,0 +1,229 @@
+	const factory = new JsonRpcProxyFactory<GitpodServer>();
+	const gitpodService = new GitpodServiceImpl<GitpodClient, GitpodServer>(factory.createProxy()) as GitpodConnection;
+
+	const webSocket = new (ReconnectingWebSocket as any)(gitpodApiEndpoint, undefined, {
+		maxReconnectionDelay: 10000,
+		minReconnectionDelay: 1000,
+		reconnectionDelayGrowFactor: 1.3,
+		connectionTimeout: 10000,
+		maxRetries: Infinity,
+		debug: false,
+		startClosed: false,
+		WebSocket: class extends WebSocket {
+			constructor(address: string, protocols?: string | string[]) {
+				super(address, protocols, {
+					headers: {
+						'Origin': new URL(gitpodHost).origin,
+						'Authorization': `Bearer ${serverToken}`
+					}
+				});
+			}
+		}
+	});
+	webSocket.onerror = console.error;
+	doListen({
+		webSocket: webSocket as any,
+		onConnection: connection => factory.listen(connection),
+		logger: new ConsoleLogger()
+	});
+
+	return gitpodService;
+}
+
+export class GitpodInsightsAppender implements ITelemetryAppender {
+
+	private _asyncAIClient: Promise<GitpodConnection> | null;
+	private _defaultData: { [key: string]: any } = Object.create(null);
+
+	constructor() {
+		this._asyncAIClient = null;
+	}
+
+	private _withAIClient(callback: (aiClient: Pick<GitpodServer, 'trackEvent'>) => void): void {
+		if (!this._asyncAIClient) {
+			this._asyncAIClient = getSupervisorData().then(
+				(supervisorData) => {
+					this._defaultData['workspaceId'] = supervisorData.workspaceId;
+					this._defaultData['workspaceInstanceId'] = supervisorData.instanceId;
+
+					return getClient(supervisorData.serverToken, supervisorData.gitpodHost, supervisorData.gitpodApiEndpoint);
+				}
+			);
+		}
+
+		this._asyncAIClient.then(
+			(aiClient) => {
+				callback(aiClient.server);
+			},
+			(err) => {
+				onUnexpectedError(err);
+				console.error(err);
+			}
+		);
+	}
+
+	log(eventName: string, data?: any): void {
+		this._withAIClient((aiClient) => {
+			data = mixin(data, this._defaultData);
+			data = validateTelemetryData(data);
+			const gitpodTrackEvent = mapTelemetryData(eventName, data.properties);
+			if (gitpodTrackEvent) {
+				aiClient.trackEvent(gitpodTrackEvent);
+			}
+		});
+	}
+
+	flush(): Promise<any> {
+		return Promise.resolve(undefined);
+	}
+}
+
+const formatEventName = (str: string) => {
+	return str
+		.replace(/^[A-Z]/g, letter => letter.toLowerCase())
+		.replace(/[A-Z]/g, letter => `_${letter.toLowerCase()}`)
+		.replace(/[^\w]/g, '_');
+};
+
+const filterEvent = new Set<string>();
+function mapTelemetryData(eventName: string, data: any): RemoteTrackMessage | undefined {
+	if (filterEvent.has(eventName)) {
+		return undefined;
+	}
+
+	switch (eventName) {
+		case 'editorOpened':
+			if ((<string>data.typeId) !== 'workbench.editors.files.fileEditorInput') {
+				return undefined;
+			}
+			filterEvent.add(eventName);
+			return {
+				event: `vscode_${formatEventName(eventName)}`,
```

Comment:
it should be  `vscode_file_access` with `read` kind.

I think it would be better to wait for approval with this PR.

We should be careful with adding new events. Because we need to track them in these doc and make sure that they make sense for @loujaybee and @jakobhero: https://www.notion.so/gitpod/ide_configuration_changed-8c5c75d33a204536bba734f337e48e56

Reply:
I see, sorry about that, do you want me to revert it? or just push the changes directly?

Reply:
It should be enough to make a new PR which properly collects `vscode_file_access`. And merge it only after approval this time.